app/controllers/registers_controller.rb

```diff
@@ -25,6 +28,25 @@ def show
     @feedback = Feedback.new
   end
 
+  def field_link_resolver(field, field_value, register_slug = @register.slug)
+    single_resolver = lambda { |f, fv|
+      if f['register'].present? && f['register'] != register_slug
```

Comment:
This will prevent linking in any register that links to itself. See `allergen` register, `allergen-group` field - this field is a link but not resolving.

Perhaps the logic should check whether the field has the same name as the register rather than whether the `register` property has the same name.